NitroxClient/Persistence/PersistedClientData.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.IO;
+using NitroxClient.Persistence.Model;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+using NitroxModel.Logger;
+
+namespace NitroxClient.Persistence
+{
+    public static class PersistedClientData
+    {
+        private static PersistedClientDataModel model = null;
+
+        private const string FILE_NAME = ".\\client.json";
+
+        private static void CheckSingleton()
+        {
+            if (model == null)
+            {
+                string text = File.ReadAllText(FILE_NAME);
+                model = JsonConvert.DeserializeObject<PersistedClientDataModel>(text);
+            }
+        }
+
+        public static Guid GetToken(string ip, string port)
+        {
+            CheckSingleton();
+            foreach (SavedServer pair in model.SavedServers)
+            {
+                if (pair.Ip.Equals(ip) && pair.Port.Equals(port)) //Always use trusted members to the left a .Equals call. Don't compare potentially unsafe data; assume parameters might be null.
+                {
+                    return pair.Token;
+                }
+            }
+            return Guid.Empty;
+        }
+
+        public static void EmplaceServer(string name, string ip, string port, Guid token)
+        {
+            CheckSingleton();
+            SavedServer savedServer = new SavedServer() { Name = name, Ip = ip, Port = port, Token = token };
+            if(GetToken(savedServer.Ip, savedServer.Port) != Guid.Empty) //Some validation is better than no validation.
+            {
+                throw new Exception($"Failed to add '{ip}':'{port}' because there is already an entry for it.");
+            }
+            model.SavedServers.Add(savedServer);
+            File.WriteAllText(JsonConvert.SerializeObject(model), FILE_NAME);
```

Comment:
The parameters are backwards.

Reply:
I have tried 

`File.WriteAllText(".\\client.json", "Test");`

But I still can't get the client to write a dang file to the disk! This is beyond frustrating!

Reply:
Fuck my life, yo. Just fuck it. So @Jannify was right, there seems to be a problem with Newtonsoft running inside the client. As soon as you comment out the Newtonsoft code from my singleton class, the client is able to write a file to the disk. Can someone please explain that behavior. That makes no sense to me AT ALL!

Reply:
https://youtu.be/yf_17TYAPbI

I seriously need a break.

Reply:
https://litjson.net/docs/quickstart/mapping-json-to-objects

I found the same behaviour occurs when you swap out Newtonsoft JSON for LitJson. The client isn't able to write files to the disk anymore. I am giving up for now.

Reply:
This this needs to be (kinda) encrypted anyway on release, we should just instead of relying on JSON, maybe use ProtoBuf exclusively, but this will make testing and development an absolute nightmare.

Reply:
I tried protbuf, same behavior. I give up.